src/main/java/delete_post/drivers/DeletePostDataAccess.java

```diff
@@ -1,25 +1,127 @@
-//package delete_post.drivers;
-//
-//import delete_post.use_case.DeletePostDsGateway;
-//
-//import java.util.List;
-//
-//public class DeletePostDataAccess implements DeletePostDsGateway{
-//
-//    private final String path;
-//
-//    public DeletePostDataAccess(String path){
-//        this.path = path;
-//    }
-//
-//    @Override
-//    public void removeFavourite(int postId, int userId){
-//
-//    }
-//    public void removeTag(int postId, String tag){
-//
-//    }
-//    public void deletePost(int postId){
-//
-//    }
-//}
\ No newline at end of file
+
+package delete_post.drivers;
+
+import delete_post.use_case.DeletePostDsGateway;
+import delete_post.use_case.PostReaderInterface;
+import entities.Post;
+
+import com.opencsv.CSVReader;
+import com.opencsv.CSVWriter;
+import com.opencsv.CSVReaderBuilder;
+import com.opencsv.exceptions.CsvException;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DeletePostDataAccess implements DeletePostDsGateway{
+
+    private final PostReaderInterface postReader;
+    private final String postPath;
+    private final String userPath;
+    private final String commentPath;
+
+    public DeletePostDataAccess(String postPath, String userPath, String commentPath,
+                                PostReaderInterface postReader){
+        this.postPath = postPath;
+        this.userPath = userPath;
+        this.commentPath = commentPath;
+        this.postReader = postReader;
+    }
+
+    @Override
+    public Post getPost(int postId){
+        List<String[]> postData = readFile(postPath);
+        for (String[] row : postData) {
+            if (Integer.parseInt(row[0]) == postId) {
+                return postReader.readPost(row);
+            }
+        }
+        System.out.println("Could not find post in database");
+        return null;
```

Comment:
this could be a problem as it will cause errors in the program if you can't find the post for some reason. If there is time, you could implement an error popup that is displayed when the post cannot be found. ViewPost has exactly the same thing.